File: atcoder.jp/contests/abc212/tasks/abc212_b/main.py
from itertools import tee
from typing import Iterable, List, TypeVar

_T = TypeVar("_T")


def pairwise(iterable: Iterable[_T]) -> "zip[tuple[_T, _T]]":
    # https://docs.python.org/ja/3/library/itertools.html#itertools.pairwise
    a, b = tee(iterable)
    next(b, None)
    return zip(a, b)


def is_weak1(xs: List[int]) -> bool:
    return len(set(xs)) == 1


def is_weak2(xs: List[int]) -> bool:
    return all((x + 1) % 10 == y for x, y in pairwise(xs))


XS = list(map(int, input()))
print("Weak" if is_weak1(XS) or is_weak2(XS) else "Strong")

File: atcoder.jp/contests/abc212/tasks/abc212_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const pairwise = <T>(array: T[]) => zip(array, array.slice(1));
const zip: {
  (a: string, b: string): [string, string][];
  <B>(a: string, b: B[]): [string, B][];
  <A>(a: A[], b: string): [A, string][];
  <A, B>(a: A[], b: B[]): [A, B][];
} = <A, B>(a: string | A[], b: string | B[]) =>
  typeof a === "string" && typeof b === "string"
    ? [...Array(Math.min(a.length, b.length))].map(
        (_, i) => [a[i], b[i]] as [string, string]
      )
    : typeof a === "string" && typeof b !== "string"
    ? [...Array(Math.min(a.length, b.length))].map(
        (_, i) => [a[i], b[i]] as [string, B]
      )
    : typeof a !== "string" && typeof b === "string"
    ? [...Array(Math.min(a.length, b.length))].map(
        (_, i) => [a[i], b[i]] as [A, string]
      )
    : [...Array(Math.min(a.length, b.length))].map(
        (_, i) => [a[i], b[i]] as [A, B]
      );

const isWeak1 = (digits: number[]) => new Set(digits).size === 1;
const isWeak2 = (digits: number[]) =>
  pairwise(digits).every(([x, y]) => (x + 1) % 10 === y);

const XS = input().split("").map(Number);
console.log(isWeak1(XS) || isWeak2(XS) ? "Weak" : "Strong");
